### PR TITLE
Add pipeline to build vhdx image.

### PR DESCRIPTION
**Pull request by liushilongbuaa**
#### Why I did it
Add VHDX image for vs platform.
#### How I did it
Add job only on master branch. Don't need to merge to release branch.
Triggered by Azure.sonic-buildimage.official.vs
#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

File: .azure-pipelines/vhdx.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  container: slave
+  steps:
+    - checkout: none
+    - download: official
+      artifact: sonic-buildimage.vs
+      patterns: 'target/sonic-vs.img.gz'
+    - script: |
+        set -ex
+        cd ../official/sonic-buildimage.vs/target/
+        gzip -d sonic-vs.img.gz
```
Comment (xumia) [inline .azure-pipelines/vhdx.yml:34]: Can we add the image in the vs official build?

File: .azure-pipelines/vhdx.yml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+trigger: none
+pr: none
+
+resources:
+  pipelines:
+    - pipeline: official
+      source: 'Azure.sonic-buildimage.official.vs'
+      trigger:
+        branches:
+        - master
```
Comment (xumia) [inline .azure-pipelines/vhdx.yml:15]: Only for master branch, is it enough?

File: .azure-pipelines/vhdx.yml (hunk 1 above)
Comment (liushilongbuaa) [reply .azure-pipelines/vhdx.yml:34]: I am not sure if it is enough. The old job is more complex.
The concern is maybe we will add a long if codes in that build script.

**Review by andyivanov — APPROVED** [review]

File: .azure-pipelines/azure-pipelines-build.yml [hunk 3]
```diff
@@ -104,6 +104,9 @@ jobs:
               make $BUILD_OPTIONS INSTALL_DEBUG_TOOLS=y target/sonic-vs.img.gz && mv target/sonic-vs.img.gz target/sonic-vs-dbg.img.gz
             fi
             make $BUILD_OPTIONS target/docker-sonic-vs.gz target/sonic-vs.img.gz target/docker-ptf.gz
+            gzip -kd target/sonic-vs.img.gz
+            docker run --rm -v "`pwd`:/sonic" -w "/sonic" sonicdev-microsoft.azurecr.io:443/sonic-slave-buster qemu-img convert target/sonic-vs.img -O vhdx -o subformat=dynamic target/sonic-vs.vhdx
```
Comment (xumia) [inline .azure-pipelines/azure-pipelines-build.yml:108]: How about change to 
NOJESSIE=y NOSTRETCH=y make sonic-slave-run emu-img convert target/sonic-vs.img -O vhdx -o subformat=dynamic target/sonic-vs.vhdx

**Comment by xumia** [comment]
@liushilongbuaa 
Is it no use in PR build, right? Do we need to disable it in pr build?
FYI: https://docs.microsoft.com/en-us/azure/devops/pipelines/build/variables?view=azure-devops&tabs=yaml
$(Build.Reason) != PullRequest

**Comment by liushilongbuaa** [comment]
> build, right? Do we need to disable

Yes, maybe we should only enable it only in master branch official build.

**Review by xumia — APPROVED** [review]

**Pull request merged**